Add time.has_time documentation

ashmckenzie opened the pull request:
## Description:

This PR adds basic documentation for the `time.has_time` condition.

**Related issue (if applicable):** N/A

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1255

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

ashmckenzie committed:
Add time.has_time documentation

ashmckenzie commented:
cc @glmnet 🙂

glmnet commented:
Content is ok, but this PR should be created against `next` branch which is for new changes.

Also consider adding a link to the new condition in the *All conditions* list here https://github.com/esphome/esphome-docs/blob/current/guides/automations.rst

ashmckenzie commented:
Thanks @glmnet, I've changed the base branch to `next` and updated https://github.com/esphome/esphome-docs/blob/529da231e511402fbb520abdd25412edd25fe51d/guides/automations.rst 🙂 May you please take another look? 🙏

OttoWinter commented on guides/automations.rst:
```diff
@@ -386,7 +386,7 @@ All Conditions
 - :ref:`switch.is_on <switch-is_on_condition>` / :ref:`switch.is_off <switch-is_off_condition>`
 - :ref:`sensor.in_range <sensor-in_range_condition>`
 - :ref:`wifi.connected <wifi-connected_condition>` / :ref:`api.connected <api-connected_condition>`
```
```suggestion
- :ref:`wifi.connected <wifi-connected_condition>` / :ref:`api.connected <api-connected_condition>`
  / :ref:`mqtt.connected <mqtt-connected_condition>`
```

I'm guessing this is because of a bad merge to current branch? Anyway, please re-add this line.

ashmckenzie replied:
Really sorry about that, fixed 🙂

glmnet commented:
Thanks!
